usr/src/cmd/mknod/mknod.c

```diff
@@ -98,13 +98,13 @@ main(int argc, char **argv)
 			usage();
 		}
 		majno = (major_t)number(argv[3]);
-		if (majno == (major_t)-1 || majno > MAXMAJ) {
+		if (majno == (major_t)-1 || majno > (major_t)MAXMAJ) {
 			(void) fprintf(stderr, "mknod: invalid major number "
 			    "'%s' - valid range is 0-%lu\n", argv[3], MAXMAJ);
 			return (2);
 		}
 		minno = (minor_t)number(argv[4]);
-		if (minno == (minor_t)-1 || minno > MAXMIN) {
+		if (minno == (minor_t)-1 || minno > (minor_t)MAXMIN) {
```

Comment:
I know what's going on with this and MAXMAJ, but I'd like to talk to some people smarter than me about it, because this will allow mknod for the first time (ever?) to create devices that 32bit software can't interpret.

Reply:
It would probably be best to change this to check against `MAXMAJ32` and `MAXMIN32` to ensure we don't create devices other software can't interpret, at least for now.  Rather than giving it the full range and possibly causing chaos for software trying to use the resulting nodes.

Reply:
Ok done.